cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_best_split_sp_opt_impl_dpc.cpp

```diff
@@ -0,0 +1,677 @@
+                    const Index node_idx = item.get_global_id(1) / ftr_worker_per_node_count;
+                    if ((processed_node_cnt + node_idx) > (node_count - 1) ||
+                        ftr_group_id > (selected_ftr_count - 1)) {
+                        return;
+                    }
+
+                    const Index node_id = node_indices_ptr[node_ind_ofs + node_idx];
+                    Index* node_ptr = node_list_ptr + node_id * impl_const_t::node_prop_count_;
+
+                    const Index local_id = item.get_local_id(0);
+
+                    const Index sub_group_id = sbg.get_group_id();
+                    const Index sub_group_local_id = sbg.get_local_id();
+
+                    const Index row_ofs = node_ptr[impl_const_t::ind_ofs];
+                    const Index row_count = node_ptr[impl_const_t::ind_lrc];
+
+                    split_smp_t sp_hlp;
+                    split_info<Float, Index, Task> bs;
+
+                    // slm pointers declaration
+                    byte_t* local_byte_buf_ptr = local_byte_buf.get_pointer().get();
+                    hist_type_t* local_hist_buf_ptr =
+                        get_buf_ptr<hist_type_t>(&local_byte_buf_ptr, local_hist_buf_size);
+                    Float* local_buf_float_ptr =
+                        get_buf_ptr<Float>(&local_byte_buf_ptr, local_buf_float_size);
+
+                    bs.init_clear(item, local_hist_buf_ptr + 0 * hist_prop_count, hist_prop_count);
+
+                    Index processed_ftr_count = 0;
+                    for (Index ftr_idx = ftr_group_id; ftr_idx < selected_ftr_count;
+                         ftr_idx += ftr_worker_per_node_count) {
+                        processed_ftr_count++;
+
+                        split_info<Float, Index, Task> ts;
+                        ts.init(local_hist_buf_ptr + 1 * hist_prop_count, hist_prop_count);
+                        ts.ftr_id = selected_ftr_list_ptr[node_id * selected_ftr_count + ftr_idx];
+                        Index i = local_id;
+                        Index id = (i < row_count) ? tree_order_ptr[row_ofs + i] : index_max;
+                        Index bin =
+                            (i < row_count) ? data_ptr[id * column_count + ts.ftr_id] : index_max;
+                        Float response = (i < row_count) ? response_ptr[id] : Float(0);
+                        Index response_int =
+                            (i < row_count) ? static_cast<Index>(response) : Index(-1);
+
+                        ts.ftr_bin = Index(-1);
+
+                        ts.ftr_bin = sycl::reduce_over_group(item.get_group(),
+                                                             bin > ts.ftr_bin ? bin : index_max,
+                                                             minimum<Index>());
+                        while (ts.ftr_bin < index_max) {
+                            const Index count = (bin <= ts.ftr_bin) ? 1 : 0;
```

Comment:
On line 239 there is an Index(-1) initialization. Maybe Index(1) : Index(0)?

Reply:
thank you, replaced with just -1